### DISCUSSION TITLE
A few questions about the Tpot CE

### DISCUSSION THREAD

**Discussion by 0hex7** [telekom-security/tpotce#1421] [Q&A] [answered]
I am a newbie to this honeypot concept. I have learnt a few basic things and now i feel like i know what i am doing with the honeypot.
However, I have a few questions to the you all,
1. by default so many honeypots are running on the tpot ce once its installed as a standard installation in the computer as a standalone system. so when an attacker does nmap against the ip of the tpot system, obviously he will see that there are around 2000 ports open. and by looking at them one can obviously make out that its a honeypot.
now my question is how can we disable any honeypot that snot needed? like, if i want to use and run only cowrie in my tpot and stop all other honeypots, how do i do that?
2. Another question is what if i want to implement some lures or any other Mitre engage affect goal approaches like, software manipulation or introducing vulnerabilities to lure the attacker in.
 in such cases what should i do?
  how should i proceed?
  do i need to edit the entire source code or any other options are available.
  
  i will be eternally grateful if you can help me out with this.
  thanks a lottttt
![Screenshot 2023-10-18 144003](https://github.com/telekom-security/tpotce/assets/108691415/371e5b53-1c38-428e-ae36-ee5fdc786e4e)
![Screenshot 2023-10-18 144026](https://github.com/telekom-security/tpotce/assets/108691415/527590c2-ebe5-4823-ab74-8ae8af3ae267)

**Comment by Ganesh-403** [comment, marked as answer]
@0hex7 
Welcome to T-Pot and honeypots! These are great questions, and fortunately T-Pot already provides mechanisms for most of what you're trying to achieve—you generally **don't need to modify the source code**.

### 1. Can I run only Cowrie (or a subset of honeypots)?

Yes.

T-Pot is built around Docker Compose, and the project officially supports customizing which honeypots and services are deployed.

You have two options:

- **Use one of the predefined compose files** (e.g. `mini.yml`, `sensor.yml`, etc.), or
- **Generate your own configuration** using the built-in `customizer.py` script, which lets you select only the honeypots and tools you want. The documentation also notes that it checks for port conflicts while generating the compose file. :contentReference[oaicite:0]{index=0}

The general workflow is:

```bash
cd ~/tpotce/compose
python3 customizer.py
```

Then replace the active compose file:

```bash
sudo systemctl stop tpot
cp docker-compose-custom.yml ~/tpotce/docker-compose.yml
sudo systemctl start tpot
```

If your goal is to expose only Cowrie, this is the recommended approach instead of simply stopping containers manually, since the compose file becomes the source of truth. :contentReference[oaicite:1]{index=1}

---

### 2. Won't attackers recognize T-Pot because many ports are open?

Potentially, yes.

A large number of exposed services can indicate that the target is a honeypot. Whether that's a problem depends on your objective:

- If you're collecting **broad internet scanning activity**, exposing many services is perfectly reasonable.
- If you're trying to emulate a **specific production server**, exposing only the services that fit that host (for example, SSH with Cowrie) is generally more believable.

Using a custom compose file to deploy only the services you need helps reduce this fingerprint.

---

### 3. What about MITRE ENGAGE techniques, lures, or fake vulnerabilities?

This depends on what you're trying to achieve.

Many honeypots already include configurable deception features (banners, filesystems, fake credentials, responses, etc.), so you often don't need to edit the application source code.

For example:

- Cowrie supports customizing its emulated filesystem, SSH banner, users, and command responses.
- Other honeypots have their own configuration files that can be overridden through Docker volumes.

If you want to create more advanced deception (e.g. fake applications, vulnerable web pages, custom documents, or software manipulation scenarios), that typically involves **customizing the individual honeypot's configuration** or adding your own services alongside T-Pot, rather than modifying T-Pot itself.

---

### 4. Should I edit T-Pot's source code?

In most cases, **no**.

From what I can tell, T-Pot is intended to orchestrate and manage the honeypots rather than be the place where you implement custom deception logic. The recommended customization points are:

- selecting which honeypots to deploy,
- editing the Docker Compose configuration,
- configuring the individual honeypots,
- adding your own Docker services if needed. :contentReference[oaicite:2]{index=2}

One question that would help provide more targeted advice:

- Are you building this for **research/learning**, **enterprise threat detection**, or as part of a **MITRE ENGAGE/deception project**? The best architecture can differ depending on your goal.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.

**Reply by 0hex7** [reply]
Thanks, man. Its been 3 years, but thanks for the answers, appreciate it

**Reply by Ganesh-403** [reply]
haha
